WorkingInR/3_Importing_Data_in_R.Rmd

```diff
@@ -2,61 +2,252 @@
++ software-based preferences, the settings which are defined _by default_ by the software that you have used to punch your data,
++ user-based preferences, the settings that _you_ choose when punching or saving data.
+
+It is thus very important to know what your CSV file is made of.
+We therefore recommend to systematically inspect your data file before importing in R.
+One way to do it is simply to open the file in a text editor (or visualize it in RStudio) and define:
+
++ which symbol is used as decimal separator (`,` or `.`),
++ which symbol is used as delimiter (`,` or `;`).
+
+
+# The package Readr
+
+`readr` is a package which is part of the `tidyverse` family.
+`readr` comes with several functions that can import your data file. 
+Among them are:
+
++ `read_delim()`,
++ `read_csv()`,
++ `read_csv2()`.
+
+We will focus essentially on `read_delim()` as it can import all the formats. 
+`read_csv()` and `read_csv2()` are simplified version of `read_delim()` which are preset to import CSV files in which a comma `,` or a semi-colon `;` is defined as delimiter, respectively.
+
+
+## read_delim()
 
-# Foreword
+`read_delim()` is a simple function that does several things at once. First, it finds and reads the file to import. Then it looks at the first 1000 rows of the file and guesses the number of variables and rows, and the data type for each variable. Finally, it creates a *tibble* and puts the data in it while keeping the order of both the columns and rows and the nature of the data.
 
-In this chapter, you will learn to import data from file into an R object. The R base package and several packages come with a set of functions that can do the job. Here, we will focus on the functions which are part of the package `tidyverse`. So, if it is not done yet, install the package with `install.packages("tidyverse")` and/or load it with `library(tidyverse)`.
+In its simplest form, `read_delim()` only needs to know two things about your data file:
 
-# Five functions to import data from a file
++ where to find it,
++ which delimiter to consider.
 
-Here, you will learn about 5 functions: `read_csv()`, `read_csv2()`, `read_excel()`, `read_table()` and `read_fwd()`. All these functions have the same purpose, namely to:
+The argument to locate the file on your machine is `file = " "`, and the argument to indicate the delimiter is `delim = " "`. 
 
-1. read a file,
-2. retrieve the data, 
-3. place the data properly into a data frame.
+In the following example, we import the file `weather.csv` located in the subfolder `Files` of the current RStudio project into the object `weather`. In `weather.csv`, the delimiter is `,` and the decimal separator is `.`
 
-The main difference is that each of these functions is dedicated to reading one specific type of file.
+```{r delim-csv, echo=TRUE} 
+weather <- read_delim(file = "Files/weather.csv", delim = ",")
+```
```

Comment:
Can the data be in a folder called data and not files?

Maybe this is also a good place to explain the advantage of using a RProject again.